lib/dom-actions.ts

```diff
@@ -157,4 +157,21 @@ export class DomActions implements DomActionsProvider {
         }
         return this.querySelectorChain(selector);
     }
+
+    waitForMutation(selector: ElementSelector): Promise<boolean> {
+        const node = this.elementSelector(selector);
+        if (node.length === 0) {
+            throw new Error(`${selector} did not match any elements`);
+        }
+        return new Promise((resolve) => {
+            const observer = new MutationObserver(() => {
+                observer.disconnect();
```

Comment:
If the DOM doesn’t mutate then this observer won’t be cleaned up (memory leak) and `findCmp` will hang indefinitely. Maybe add a `timeout` param and have `waitForMutation` throw an error when it times out? Then `findCmp` can catch the error and stop retrying. Can use a very long timeout e.g. 1 minute.

Reply:
Good call - I added this behavior now: after 60s the promise will be rejected, and `findCmp` will see that rejection and exit early.